reservation_units/utils/reservation_unit_pricing_helper.py

```diff
@@ -22,11 +22,24 @@ def get_future_price(cls, reservation_unit: ReservationUnit) -> ReservationUnitP
     def get_price_by_date(
         cls, reservation_unit: ReservationUnit, by_date: datetime.date
     ) -> ReservationUnitPricing | None:
+        active_price = cls.get_active_price(reservation_unit)
         future_price = cls.get_future_price(reservation_unit)
-        if future_price and future_price.begins <= by_date:
+
+        if (
+            future_price
+            and future_price.begins <= by_date
+            and (
+                # Only return future price if it has begun and has the same tax percentage as the current active price.
+                # When the future price has a different tax percentage, it should only be used for reservations which
+                # are made after the ReservationUnitPricing.begins date. TILA-3470
+                active_price.tax_percentage == future_price.tax_percentage
+                # If either of the prices is free, the future price can be returned, as the percentage is irrelevant.
+                or active_price.pricing_type != future_price.pricing_type
```

Comment:
I'm wondering if there is still an edge case here: Current pricing is FREE and the future pricing is PAID, and _after_ the tax percentage change, BUT there is another PAID pricing _before_ the tax percentage change. In this case, should we use the PAID pricing _before_ the tax percentage change, as that would otherwise be the price at that time?

Likely, this won't happen.

Reply:
This is a rare case, but still a good point.
This could be better handled when reworking how pricing statuses work, added a note for myself :)